File: docs/articles/reading-notes/head-first-design-pattern/code/01-strategy-pattern/02-SimUDuck/02-SimUDuck/Duck.m
//
//  Duck.m
//  01-SimUDuck
//
//  Created by zhouke on 2018/6/18.
//  Copyright © 2018年 zhouke. All rights reserved.
//

#import "Duck.h"

@implementation Duck

- (void)performQuack
{
    [self.quackBehavior qucak];
}

- (void)performFly
{
    [self.flyBehavior fly];
}

- (void)swim
{
    NSLog(@"swim");
}

- (void)display
{
    NSLog(@"display");
}


@end

File: docs/articles/reading-notes/head-first-design-pattern/code/07-adapter-pattern/01-Duck/01-Duck/Duck.swift
//
//  Duck.swift
//  01-Duck
//
//  Created by zhouke on 2018/7/7.
//  Copyright © 2018年 zhouke. All rights reserved.
//

import Foundation

protocol Duck {
    func quack()
    func fly()
}

extension Duck {
    func quack() {
        print("Quack")
    }
    
    func fly() {
        print("I'm flying")
    }
}
